### DISCUSSION TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'out/build-info'

### DISCUSSION THREAD

**Discussion by joedakwa** [crytic/slither#1787] [Q&A]
### Describe the issue:

I have recently pulled down a repo from GH and i am trying to run basic slither commands from here.

I have currently ran ```slither . slither . --solc-remaps '@openzeppelin/contracts/=node_modules/@openzeppelin/contracts/'```

and had the below output.

'forge clean' running (wd: /home/joedakwa/joe/audits/2023-02-kairos-joedakwa/kairos-contracts)
'forge build --build-info --force' running
Compiling 112 files with 0.8.18
Solc 0.8.18 finished in 26.66s
Compiler run successful

Traceback (most recent call last):
  File "/home/joedakwa/.local/lib/python3.10/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/foundry.py", line 90, in compile
    hardhat_like_parsing(crytic_compile, self._target, build_directory, self._target)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/hardhat.py", line 49, in hardhat_like_parsing
    os.listdir(build_directory), key=lambda x: os.path.getmtime(Path(build_directory, x))
FileNotFoundError: [Errno 2] No such file or directory: 'out/build-info'
Error in .
Traceback (most recent call last):
  File "/home/joedakwa/.local/lib/python3.10/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/foundry.py", line 90, in compile
    hardhat_like_parsing(crytic_compile, self._target, build_directory, self._target)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/hardhat.py", line 49, in hardhat_like_parsing
    os.listdir(build_directory), key=lambda x: os.path.getmtime(Path(build_directory, x))
FileNotFoundError: [Errno 2] No such file or directory: 'out/build-info'

from the below directory. All the contracts are inside of src.

joe/audits/2023-02-kairos-joedakwa/kairos-contracts/

I tried also running this command:

slither . --solc-remaps '@openzeppelin/contracts/=node_modules/@openzeppelin/contracts/'

from joe/audits/2023-02-kairos-joedakwa/kairos-contracts/src

(src is where the contracts are)

and had the below output.

I have a remappings json file currently looking like this:

```
{
  "remappings": [
    "@openzeppelin/contracts/token/ERC20/=node_modules/@openzeppelin/contracts/token/ERC20/"
  ]
}
```

Error: Source "node_modules/@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> ClaimFacet.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Any advice here?

### Code example to reproduce the issue:

N/A

### Version:

0.9.2

**Comment by 0xalpharush** [comment]
Do you have `out` configured in your foundry.toml? https://book.getfoundry.sh/reference/config/project#out 
If so, you need to pass the path  to slither with the flag `--foundry-out-directory $VALUE_SET_IN_FOUNDRY_TOML`. You shouldn't need to pass anything with remappings to slither as Foundry handles that and slither just reads the artifact output

**Comment by joedakwa** [comment]
> Do you have `out` configured in your foundry.toml? https://book.getfoundry.sh/reference/config/project#out If so, you need to pass the path to slither with the flag `--foundry-out-directory $VALUE_SET_IN_FOUNDRY_TOML`. You shouldn't need to pass anything with remappings to slither as Foundry handles that and slither just reads the artifact output

There is a remappings.txt file in the directory.

```
@openzeppelin/=./node_modules/@openzeppelin/
forge-std/=./node_modules/forge-std/src/
ds-test/=./node_modules/ds-test/src/
diamond/=./node_modules/diamond/
```

This is currently in my foundry.toml

```
[profile.default]
src = 'src'
test = 'test'
out = 'out/foundry'
cache_path = 'out/foundry-cache'
broadcast = 'out/foundry-broadcast'
fs_permissions = [{ access = "write", path = "./out/"}]
allow_paths= ['./node_modules']

[profile.doc]
out = 'out/docs'
# repository = fixme add public contracts repo
```

What is the correct slither command in this instance then ser?

**Comment by 0xalpharush** [comment]
The out directory has been modified to `out = 'out/foundry'`. I believe `slither . --foundry-out-directory out/foundry` should work. By default, it is just `out`

**Comment by joedakwa** [comment]
> slither . --foundry-out-directory out/foundry

I still get

Compilation warnings/errors on ./ClaimFacet.sol:
Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> ClaimFacet.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

When running that command

**Comment by joedakwa** [comment]
> The out directory has been modified to `out = 'out/foundry'`. I believe `slither . --foundry-out-directory out/foundry` should work. By default, it is just `out`

I still get

Compilation warnings/errors on ./ClaimFacet.sol:
Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
--> ClaimFacet.sol:4:1:
|
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

When running that command

**Comment by 0xalpharush** [comment]
This is an error from the compiler and not slither. If `forge build` is failing, slither will not work. It looks like you're missing a dependency and likely need to run npm/ yarn install depending on the project's setup.

**Comment by joedakwa** [comment]
> This is an error from the compiler and not slither. If `forge build` is failing, slither will not work. It looks like you're missing a dependency and likely need to run npm/ yarn install depending on the project's setup.

I just checkout package.json

```
{
  "name": "@kairos-loan/contracts",
  "version": "0.1.0",
  "scripts": {
    "build": "yarn build:foundry",
    "build:foundry": "forge build --extra-output-files abi",
    "start": "anvil --silent --block-time 1 > out/anvil-logs.txt & yarn deploy:local &",
    "lint": "solhint './src/**/*.sol ./scripts/sol/**/*.sol' -c ../../.solhint.json && eslint ./scripts/ts/**/*.ts",
    "lint:fix": "solhint './src/**/*.sol ./scripts/sol/**/*.sol' -c ../../.solhint.json --fix && eslint ./scripts/ts/**/*.ts --fix",
    "prettify": "prettier --write ./**/*.sol ./scripts/ts/**/*.ts",
    "test": "forge test",
    "generate:doc": "forge doc --build",
    "generate:coverage": "forge coverage --report lcov"
  },
  "dependencies": {
    "@openzeppelin/contracts": "^4.6.0",
    "diamond": "https://github.com/mudgen/diamond-1-hardhat",
    "dotenv": "^16.0.3",
    "ds-test": "https://github.com/dapphub/ds-test",
    "ethers": "^5.7.2",
    "forge-std": "https://github.com/foundry-rs/forge-std#master",
    "solmate": "^6.5.0"
  }
```

I tried NPM i, but the command still didn't work, not sure if the above is an indication of what i am missing?

**Comment by elopez** [comment]
did you run `slither . --foundry-out-directory out/foundry` from the directory where `foundry.toml` is located? can you post the full execution log?

**Comment by joedakwa** [comment]
> did you run `slither . --foundry-out-directory out/foundry` from the directory where `foundry.toml` is located? can you post the full execution log?

I ran this command:

```/joe/audits/2023-02-kairos-joedakwa/kairos-contracts/src$ slither . --foundry-out-directory out/foundry```

![kairos](https://user-images.githubusercontent.com/57464255/226633593-75985ba1-ebb6-4bcc-8db7-82b41ad6bf62.jpg)

This was the execution output

Traceback (most recent call last):
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/solc.py", line 559, in _run_solc
    ret: Dict = json.loads(stdout)
  File "/usr/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/joedakwa/.local/lib/python3.10/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 641, in compile_all
    compilations.append(CryticCompile(filename, **kwargs))
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/solc.py", line 154, in compile
    targets_json = _get_targets_json(compilation_unit, self._target, **kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/solc.py", line 283, in _get_targets_json
    return _run_solc(
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/solc.py", line 563, in _run_solc
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> ClaimFacet.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Errors.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC721/IERC721.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Errors.sol:5:1:
  |
5 | import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Objects.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC721/IERC721.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Objects.sol:5:1:
  |
5 | import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Storage.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> utils/Erc20CheckedTransfer.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol" not found: File not found. Searched the following locations: "".
 --> SupplyPositionLogic/NFTUtils.sol:4:1:
  |
4 | import {IERC721Receiver} from "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/utils/Address.sol" not found: File not found. Searched the following locations: "".
 --> SupplyPositionLogic/NFTUtils.sol:5:1:
  |
5 | import {Address} from "@openzeppelin/contracts/utils/Address.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


Error in .
Traceback (most recent call last):
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/solc.py", line 559, in _run_solc
    ret: Dict = json.loads(stdout)
  File "/usr/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/joedakwa/.local/lib/python3.10/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 641, in compile_all
    compilations.append(CryticCompile(filename, **kwargs))
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/solc.py", line 154, in compile
    targets_json = _get_targets_json(compilation_unit, self._target, **kwargs)
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/solc.py", line 283, in _get_targets_json
    return _run_solc(
  File "/home/joedakwa/.local/lib/python3.10/site-packages/crytic_compile/platform/solc.py", line 563, in _run_solc
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> ClaimFacet.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Errors.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC721/IERC721.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Errors.sol:5:1:
  |
5 | import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Objects.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC721/IERC721.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Objects.sol:5:1:
  |
5 | import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> DataStructure/Storage.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC20/IERC20.sol" not found: File not found. Searched the following locations: "".
 --> utils/Erc20CheckedTransfer.sol:4:1:
  |
4 | import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol" not found: File not found. Searched the following locations: "".
 --> SupplyPositionLogic/NFTUtils.sol:4:1:
  |
4 | import {IERC721Receiver} from "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "@openzeppelin/contracts/utils/Address.sol" not found: File not found. Searched the following locations: "".
 --> SupplyPositionLogic/NFTUtils.sol:5:1:
  |
5 | import {Address} from "@openzeppelin/contracts/utils/Address.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

**Comment by elopez** [comment]
You need to run it from the `kairos-contracts` folder (i.e. not from the `src` folder like you did, but from the project root folder, which has the `foundry.toml` file)

**Comment by joedakwa** [comment]
> You need to run it from the `kairos-contracts` folder (i.e. not from the `src` folder like you did, but from the project root folder, which has the `foundry.toml` file)

OK great this works thank you ser

**Comment by joedakwa** [comment]
> You need to run it from the `kairos-contracts` folder (i.e. not from the `src` folder like you did, but from the project root folder, which has the `foundry.toml` file)

I assume this works in a similar fashion to running Mythril?

**Comment by elopez** [comment]
I'm not familiar with how Mythril works, but slither (through crytic-compile) will detect that this is a Foundry project when it sees the `foundry.toml` file, and call `forge` for you to perform the build. If you run Slither from some other folder, it won't see it's a Foundry project and try to compile any .sol files with `solc` instead (which is usually not what you want, and will give you problems with remappings and such unless your code is self-contained or trivial).

**Comment by joedakwa** [comment]
> I'm not familiar with how Mythril works, but slither (through crytic-compile) will detect that this is a Foundry project when it sees the `foundry.toml` file, and call `forge` for you to perform the build. If you run Slither from some other folder, it won't see it's a Foundry project and try to compile any .sol files with `solc` instead (which is usually not what you want, and will give you problems with remappings and such unless your code is self-contained or trivial).

right i see, is there a separate command to run for specific sol files? Rather than running ```slither .``` ?

**Comment by elopez** [comment]
you can run e.g. `slither path/foo/file.sol`. But again, doing so will attempt to compile the file with `solc` directly instead of using `forge`, and you will need to configure remappings manually.

If you're looking to reduce the scope where detectors are run, look into the [path filtering](https://github.com/crytic/slither/wiki/Usage#path-filtering) option.

**Comment by joedakwa** [comment]
I mean ultimately i'd like to avoid creating remappings files if i can.

So ideally, i'd like to find, for example, high risk vulnerabilities, how would i do that?

 ```slither . --exclude-optimization --exclude-informational --exclude-low --foundry-out-directory out/foundry```

Like this for example?

**Comment by joedakwa** [comment]
> I mean ultimately i'd like to avoid creating remappings files if i can.
> 
> So ideally, i'd like to find, for example, high risk vulnerabilities, how would i do that?
> 
> `slither . --exclude-optimization --exclude-informational --exclude-low --foundry-out-directory out/foundry`
> 
> Like this for example?

I mean ultimately i'd like to avoid creating remappings files if i can.

So ideally, i'd like to find, for example, high risk vulnerabilities, how would i do that?

slither . --exclude-optimization --exclude-informational --exclude-low --foundry-out-directory out/foundry

Like this for example?

**Reply by elopez** [reply]
That sounds good if you're looking to get medium/high detections only 👍

**Reply by joedakwa** [reply]
> That sounds good if you're looking to get medium/high detections only 👍

Right gotcha, when i ran that full command however, it seems to still print all of the detectors still? I would be expecting just "red" outputs, as i excluded everything except high?

**Reply by joedakwa** [reply]
> Right gotcha, when i ran that full command however, it seems to still print all of the detectors still? I would be expecting just "red" outputs, as i excluded everything except high?

Right gotcha, when i ran that full command however, it seems to still print all of the detectors still? I would be expecting just "red" outputs, as i excluded everything except high?

**Reply by elopez** [reply]
ah, I believe you also need to add `--no-fail-pedantic` (`fail-pedantic` is the default, and that requires enabling optimization and informational detectors). It's a bit unintuitive, and being tracked for improvement on #1458

**Reply by joedakwa** [reply]
> ah, I believe you also need to add `--no-fail-pedantic` (`fail-pedantic` is the default, and that requires enabling optimization and informational detectors). It's a bit unintuitive, and being tracked for improvement on #1458

So something like this?

```slither . -- no-fail-pedantic --exclude-medium  --exclude-low --foundry-out-directory out/foundry```

**Reply by elopez** [reply]
keep the exclusions, but add `--no-fail-pedantic`: 

`slither . --no-fail-pedantic --exclude-optimization --exclude-informational --exclude-low --foundry-out-directory out/foundry`

**Reply by joedakwa** [reply]
> `slither . --no-fail-pedantic --exclude-optimization --exclude-informational --exclude-low --foundry-out-directory out/foundry`

Yep lovely this works :-) cheers

**Reply by joedakwa** [reply]
> keep the exclusions, but add `--no-fail-pedantic`:
> 
> `slither . --no-fail-pedantic --exclude-optimization --exclude-informational --exclude-low --foundry-out-directory out/foundry`

I just have one more question, when i run ```slither BorrowFacet.sol --print function-summary --foundry-out-directory out/foundry

I get errors similar to before. If i run ```slither . --print functions-summary --foundry-out-directory out/foundry``` it works fine.

The error message

"Compilation warnings/errors on src/BorrowFacet.sol:
Error: Source "@openzeppelin/contracts/token/ERC721/IERC721.sol" not found: File not found. Searched the following locations: "".
 --> src/BorrowFacet.sol:4:1:
  |
4 | import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Any reason as to why this is happening?

**Reply by elopez** [reply]
if you pass a standalone `.sol` file as target, Slither will _not_ use Foundry -- instead, it will invoke solc to build it, and you'll have to handle remappings manually.

**Reply by joedakwa** [reply]
> if you pass a standalone `.sol` file as target, Slither will _not_ use Foundry -- instead, it will invoke solc to build it, and you'll have to handle remappings manually.

Lovely worked it out.

```slither src/AdminFacet.sol --print function-summary --solc-remaps @openzeppelin/contracts/=node_modules/@openzeppelin/contracts/```

**Reply by joedakwa** [reply]
> if you pass a standalone `.sol` file as target, Slither will _not_ use Foundry -- instead, it will invoke solc to build it, and you'll have to handle remappings manually.

Any reason why i am getting this error?

```AttributeError: 'FunctionTopLevel' object has no attribute 'contract_declarer'```

When running ```slither src/AuctionFacet.sol --print call-graph --solc-remaps @openzeppelin/contracts/=node_modules/@openzeppelin/contracts/```

**Reply by elopez** [reply]
That looks like this issue: https://github.com/crytic/slither/issues/1437

**Reply by joedakwa** [reply]
> That looks like this issue: #1437

Looks like its still in progress, thanks though :-)
